### ISSUE TITLE
Apply test badge to payment component when it has focus

### ISSUE THREAD

**Issue by pierorocca** [Automattic/woocommerce-payments#9625]
### Description
Recently we updated the card component design to include a new test badge. Originally the badge was applied to all payment methods leading to a wall of yellow test badges that looked repetitive and highlighted that there's no universal test mode in WooCommerce.

The choice was made to add the badge only to the card component since it's the only one that wouldn't present a "test mode" message at some point in the shopper UI.

Looking back at the discussion with design about other alternatives, the option to display the badge on all payment methods but only when the payment method is selected and expanded is very feasible.0

### Acceptance criteria
1. The Test badge is eligible to be displayed on any payment methods, excluding express checkout buttons.
2. The badge is displayed if the payment method is selected and the section is expanded.
3. If the payment method is not selected, its Test badge is not displayed.
4. If the site is in WooPayments Live mode, the Test badge is never displayed.
5. There can only be zero or one test badges visible at any one time.

cc @FangedParakeet

**Comment by frosso** [comment]
@pierorocca for demo purposes - here are some examples for the shortcode checkout for a smaller width (where the "test mode" text needs wrapping) and full width

![Image](https://github.com/user-attachments/assets/a557bacf-db9b-4667-9a62-e62b58d62c72)
![Image](https://github.com/user-attachments/assets/df9a356e-e03d-401e-8fe9-480949569eb2)


Even smaller (level: impossible - the card input isn't even showing the whole number):
![Image](https://github.com/user-attachments/assets/cdc9e341-061d-49fd-93fd-5d4aba3cb56b)

**Comment by pierorocca** [comment]
Two options to try @frosso to see what looks and works best:

1: "Cards"
2: "Credit / debit card"

**Comment by pierorocca** [comment]
For level impossible, what size viewport is that?

**Comment by frosso** [comment]
> For level impossible, what size viewport is that?

@pierorocca I manually went to 270px width, just to showcase the text wrapping behavior

**Comment by frosso** [comment]
> Two options to try @frosso to see what looks and works best

@pierorocca renaming the card at checkout has further ramifications: it would need to be renamed in the settings, "order success" page, etc.

To confirm - would you like to move forward with this change?

**Comment by pierorocca** [comment]
Thanks. Unless one is on an old school flip phone I think it's safe to assume that width won't be seen in practice. What about 375px which corresponds to older form factor iphones?

**Comment by pierorocca** [comment]
Regarding the naming settings is fine not being named exactly the same thing. For order success @timur27 I believe was looking at [card brand logo] ending in 1234 for v2 and could/should apply to v1 to continue the work done adding icons for bnpl and express checkouts. What do you think?

**Comment by frosso** [comment]
> What about 375px which corresponds to older form factor iphones?

It'll show wrapped like in the second capture - this is the static version.
![Image](https://github.com/user-attachments/assets/746ed863-8dde-4b10-891a-9d7812ce80c1)

**Comment by frosso** [comment]
> Regarding the naming settings is fine not being named exactly the same thing.

Right now the codebase is made so the naming is consistent across all views - settings, onboarding, checkout, etc.
Would you like to be able to decouple the naming between the current views?

**Comment by pierorocca** [comment]
At 375px the wrapped badge doesn't look all that bad.

Order success should be decoupled as that descriptor provides no value to a shopper right?

For settings I'm indifferent. The logo next to the radio button is a good hint it means credit and debit.

**Comment by frosso** [comment]
@pierorocca to ensure it's captured correctly, could you please create a separate ticket for the renaming requirement?
I'll be AFK starting tomorrow, and I won't be able to piggyback on this existing ticket :)

**Comment by frosso** [comment]
@pierorocca an update on the current work - during the PR review process, it has been highlighted that different themes will change how the markup and the styles are applied to checkout.
In order to make things consistent across them on shortcode checkout (blocks checkout is unaffected), we'll need to make the input selection (i.e.: the radio input "circle") vertically centered, like so:
![Image](https://github.com/user-attachments/assets/ab13fb6b-f9f7-4dc4-ac16-d09c61b5e5b1)


The difference is that with the previous screenshots, the radio button was appearing right next to the payment method label.

This is pretty much consistent with the block-based checkout when new lines are displayed (albeit the block-based checkout would naturally show a larger radio circle):
![Image](https://github.com/user-attachments/assets/2e608532-0e32-4204-b2d3-1b5b3d5b80d7)

**Comment by pierorocca** [comment]
Thanks @frosso it's really hard to follow the fast moving .gif. Could you provide a screenshot at 375px  and 430px wide and desktop sized?

**Comment by pierorocca** [comment]
Credit or debit card looks offset to the right. The vertical spacing between lines makes it look like they're distinct individual lines, and correct me if I'm wrong Nikki's original designs has the logos vertically centered.

**Comment by frosso** [comment]
> Credit or debit card looks offset to the right.

I tried to alleviate the offset within our own payment methods.
WooCommerce Core will add a whitespace preceding the payment method's name, which would otherwise push the BNPL messaging.
The screenshot below is the current status:

![Image](https://github.com/user-attachments/assets/4780d4a5-2b53-485e-9531-07d6c887a328)

You'll notice that WooPayments' Klarna/Afterpay are offset, compared to WooPayment's "Credit card / debit card".
I tried to alleviate by ensuring that all WooPayments' labels are at least aligned at the same level - which makes the labels from other payment methods seem offset.

The main thing I wanted to highlight with my changes is how the radio's circle will need to be vertically centered, rather than top-aligned like it is in the current status (and Nikki's designs).
![Image](https://github.com/user-attachments/assets/3df1a352-f0a1-4c74-82aa-296474889890)

This, however, will make it consistent with the Blocks checkout, where the radio's circle is vertically aligned.

> The vertical spacing between lines makes it look like they're distinct individual lines, and correct me if I'm wrong Nikki's original designs has the logos vertically centered.

That's accurate, according to this post the icon should have been vertically centered: pdFofs-1XB-p2

Within the scope of this ticket, I have been focusing on ensuring the "Test mode" pill plays nicely with the remaining elements (i.e.: the radio circle, the payment method name, the "test mode" pill, the icon, and the BNPL messaging).
But I can spend some more time to ensure the logos are also vertically centered on shortcode.

There are multiple challenges with this ticket on shortcode, because we don't have full control over the markup (WooCommerce Core will output it in a certain way), and different themes make assumptions about the markup being present. 

I'll continue investigating a different approach.
